spec/features/donor_edits_pickup_location_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "rails_helper"
+
+feature "Donor edits pickup location" do
+  scenario "from profile" do
+    location = create(:location, :residence, :grown_on_site)
+    donor = create(:donor, location: location)
+    new_settings = {
+      location_type: "business",
+      grown_on_site: false,
+    }
+
+    visit root_path(as: donor)
+    edit_location(new_settings)
+
+    expect(page).to have_success_flash
+    expect(page).to have_setting(new_settings[:location_type])
+    expect(page).to have_setting(new_settings[:grown_on_site])
+  end
+
+  def have_success_flash
+    have_text t("locations.update.success")
+  end
+
+  def have_setting(value)
+    have_css(%{[checked][value="#{value}"]})
+  end
+
+  def edit_location(location_type: nil, grown_on_site: nil, **attributes)
+    unless location_type.nil?
+      choose option_for(:location_type, location_type)
+    end
+
+    unless grown_on_site.nil?
```

Comment:
I've been discouraged from using `unless` since I joined here. Could this be `if location_type.present?`, etc.?